odoo/addons/base/tests/test_mail.py

```diff
@@ -394,3 +395,73 @@ def test_default_email_from(self, *args):
             "The body of an email",
         )
         self.assertEqual(message["From"], "settings@example.com")
+
+    def test_get_email_from(self):
+        get_email_from = self.env['ir.mail_server']._get_email_from
+        set_param = self.env['ir.config_parameter'].set_param
+
+        # Standard case, no setting
+        set_param('mail.force.smtp.from', False)
+        set_param('mail.dynamic.smtp.from', False)
+        set_param('mail.catchall.domain', 'odoo.com')
+
+        email_from, return_path = get_email_from('admin@gmail.com')
```

Comment:
Just to avoid "real" emails (even if admin at gmail.com is probably not a real email), maybe use example.com or test.example.com ?